src/palace.pyx

```diff
@@ -133,21 +140,25 @@ def current_context() -> Optional[Context]:
     return current
 
 
-def use_context(context: Optional[Context]) -> None:
+def use_context(context: Optional[Context], thread: bool = False) -> None:
     """Make the specified context current for OpenAL operations.
 
+    If `thread` is set to `True`,  it will use a context thread.
+
     See Also
     --------
     Context : Audio environment container
     """
-    if context is None:
-        alure.Context.make_current(<alure.Context> nullptr)
+    if thread:
+        if context is None:
+            alure.Context.make_thread_current(<alure.Context> nullptr)
+        else:
+            alure.Context.make_thread_current((<Context> context).impl)
     else:
-        alure.Context.make_current((<Context> context).impl)
-
-
-# TODO: current_context_thread
-# TODO: use_context_thread
+        if context is None:
+            alure.Context.make_current(<alure.Context> nullptr)
+        else:
+            alure.Context.make_current((<Context> context).impl)
```

Comment:
Don't repeat yourself!  Cache the alure context to make the code more readable.  Also mind the docstring.